time_tracker_api/time_entries/time_entries_repository.py

```diff
@@ -181,7 +181,8 @@ def on_create(self, new_item_data: dict, event_context: EventContext):
 
     def on_update(self, updated_item_data: dict, event_context: EventContext):
         CosmosDBRepository.on_update(self, updated_item_data, event_context)
-        self.validate_data(updated_item_data, event_context)
+        if not 'deleted' in updated_item_data:
```

Comment:
Nice way to verify the status of the entry. My guessing is, where are the tests for this scenario? :P

Reply:
this "deleted" key is added to the entries to be deleted, not to the ones that are already deleted.
Tests done

Reply:
We could change to 
```
update_to_delete = 'deleted' in updated_item_data
if not update_to_delete:
```